kit/azure/buildingblocks/connectivity/buildingblock/main.tf

```diff
@@ -23,12 +28,23 @@ resource "azurerm_role_assignment" "spoke_rg" {
   scope                = azurerm_resource_group.spoke_rg.id
 }
 
+# Wait for the role assignment to become effective, azure is eventually consistent and thus sometimes flaky
+# even though the azurerm_role_assignment provider tries to cover for that, it's not always effective.
+# Example:
+# > Error: checking for presence of existing Virtual Network (Subscription: "b53caa89-ba38-4f04-a0b2-f989c98d4add"
+# > Resource Group Name: "connectivity"
+# > Virtual Network Name: "vnet-demo-app-vnet"): network.VirtualNetworksClient#Get: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="AuthorizationFailed" Message="The client 'dca119ef-092d-41ce-83d3-920d4eda271a' with object id 'dca119ef-092d-41ce-83d3-920d4eda271a' does not have authorization to perform action 'Microsoft.Network/virtualNetworks/read' over scope '/subscriptions/b53caa89-ba38-4f04-a0b2-f989c98d4add/resourceGroups/connectivity/providers/Microsoft.Network/virtualNetworks/vnet-demo-app-vnet' or the scope is invalid. If access was recently granted, please refresh your credentials."
```

Comment:
we should find a a other place for comments like that. For collie-hub its not relevant but would be part of the buildingblock

Reply:
As discussed, this is a justification for the design decision to put the sleep in (which is usually an antipattern...)